File: 1394. Find Lucky Integer in an Array/findLucky.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

int findLucky(vector<int>& arr) {
    unordered_map<int, int> mp;
    
    // Count occurrences of each number
    for (int num : arr) {
        mp[num]++;
    }
    
    int res = -1;
    
    // Find the lucky integer
    for (const auto& pair : mp) {
        if (pair.first == pair.second) {
            res = max(res, pair.first);
        }
    }
    
    return res;
}

File: 1394. Find Lucky Integer in an Array/findLucky.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var findLucky = function (arr) {
    let mp = new Map()
    let res = -1

    for (let num of arr) {
        if (!mp.has(num)) {
            mp.set(num, 0)
        }
        mp.set(num, mp.get(num) + 1)
    }

    for (let [key, val] of mp) {
        if (key === val) {
            res = Math.max(res, key)
        }
    }
    return res;
};